Parser/asdl_c.py

```diff
@@ -1437,6 +1470,8 @@ def visit(self, object):
 def generate_ast_state(module_state, f):
     f.write('struct ast_state {\n')
     f.write('    int initialized;\n')
+    f.write('    int recursion_depth;\n')
```

Comment:
Instead of the global/shared module state, would it make sense to create a per-transformation state which we would store this information and pass down on each call. I feel like this might open a can of worms if it is not properly handled (failure to reset the counter on some exit case would affect other transformations) or even race conditions? (not sure about that).

Reply:
I thought of that, but I decided to mirror how we are handled this in other places at the current moment (AST optimized and validator) mainly due to the fact that this needs to be backported and it needs to go on the next beta and I don't want big changes in all the generated code functions.

Additionally, there is some validation on exit. Also, the error mode (if we forget a code path) is very noisy as it will fail even at building time if the depth is not properly reset.

I would suggest to go with this version first so we can backport it and then porting it to your suggestion API in the main branch.

Being said that, if you feel very strongly I can give it a go now.

Reply:
Works for me, thanks for the explanation for the reasoning!